src/main/java/savings/Saving.java

```diff
@@ -0,0 +1,312 @@
+     * @param description The new goal description.
+     */
+    public void updateSavingsGoal(int index, double amount, String description) {
+        if (index >= 0 && index < savingsRecords.size()) {
+            SavingsRecord record = savingsRecords.get(index);
+            record.amount = amount;
+            record.goal = description;
+            System.out.println("I have updated your saving amount and saving goal:");
+            System.out.println((index + 1) + ". \t" + record);
+        } else {
+            System.out.println("Invalid index.");
+        }
+    }
+
+    /**
+     * Deletes the goal of a specified savings record.
+     * @param index The index of the savings record.
+     */
+    public void deleteSavingsGoal(int index) {
+        if (index >= 0 && index < savingsRecords.size()) {
+            savingsRecords.get(index).goal = " ";
+            System.out.println("I have deleted the saving goal:");
+            System.out.println((index + 1) + ". \t" + savingsRecords.get(index));
+        } else {
+            System.out.println("Invalid index.");
+        }
+    }
+
+    /**
+     * Transfers a specified amount from one savings record to another.
+     * @param fromIndex The index of the savings record to transfer from.
+     * @param toIndex The index of the savings record to transfer to.
+     * @param amount The amount to transfer.
+     */
+    public void transferSavings(int fromIndex, int toIndex, double amount) {
+        fromIndex -= 1;
+        toIndex -= 1;
+
+        // Check for valid indices
+        if (fromIndex < 0 || fromIndex >= savingsRecords.size() ||
+                toIndex < 0 || toIndex >= savingsRecords.size()) {
+            throw new IllegalArgumentException("Invalid index.");
+        }
+
+        // Prevent transferring to the same record
+        if (fromIndex == toIndex) {
+            throw new IllegalArgumentException("Cannot transfer to the same savings record.");
+        }
+
+        SavingsRecord fromRecord = savingsRecords.get(fromIndex);
+        SavingsRecord toRecord = savingsRecords.get(toIndex);
+
+        // Check for sufficient funds
+        if (fromRecord.getAmount() < amount) {
+            throw new IllegalArgumentException("Insufficient funds in the source savings.");
+        }
+
+        // Perform the transfer using setters
+        fromRecord.setAmount(fromRecord.getAmount() - amount);
+        toRecord.setAmount(toRecord.getAmount() + amount);
+
+        System.out.printf("Transferred %.2f from savings %d to savings %d.%n", amount, fromIndex + 1, toIndex + 1);
+        System.out.println("Updated records:");
+        System.out.println((fromIndex + 1) + ". \t" + fromRecord);
+        System.out.println((toIndex + 1) + ". \t" + toRecord);
+    }
+
+
+    /**
+     * Runs the savings management system, processing user commands.
+     * @param input The Scanner object for user input.
+     */
+    public void run(String input) {
```

Comment:
you can extract individual processing to different methods and improve the SLAP in this method.